bazel/envoy_build_system.bzl

```diff
@@ -297,7 +318,10 @@ def envoy_cc_test(
         name = name,
         copts = envoy_copts(repository, test = True),
         linkopts = envoy_test_linkopts(),
-        linkstatic = 1,
+        linkstatic = select({
```

Comment:
Can you factor this pattern out into a macro? I think if it's just a single `return select(...)` that Skylark is able to compose this.